diff --git a/node_modules/@google/earthengine/build/browser.js b/node_modules/@google/earthengine/build/browser.js
index 99457a7..d7a209a 100644
--- a/node_modules/@google/earthengine/build/browser.js
+++ b/node_modules/@google/earthengine/build/browser.js
@@ -26828,6 +26828,12 @@ ee.data.Profiler.Format.JSON = new ee.data.Profiler.Format("json");
   goog.global.EXPORTED_FN_INFO = exportedFnInfo;
 })();
 
-goog.global = window;
-goog.Timer.defaultTimerObject = window;
-module.exports = ee;
+// goog.global = window;
+// goog.Timer.defaultTimerObject = window;
+// module.exports = ee;
+
+// https://github.com/google/earthengine-api/pull/173
+// This change is to support running this script in a web worker,
+// which doesn't have access to the window object
+goog.Timer.defaultTimerObject = self
+module.exports = goog.global.ee = ee
